opentelemetry-userevents-exporter/src/logs/exporter.rs

```diff
@@ -0,0 +1,288 @@
+                }
+
+                if cs_b_count > 0 {
+                    eb.add_struct("PartB", cs_b_count, 0);
+                    {
+                        if level != Level::Invalid {
+                            eb.add_value("severityNumber", level.as_int(), FieldFormat::SignedInt, 0);
+                        }
+                        if is_severity_text_present {
+                            eb.add_str("severityText", log_data.record.severity_text.as_ref().unwrap().as_ref(), FieldFormat::SignedInt, 0);
+                        }
+                        if is_body_present {
+                            eb.add_str("body",
+                    match log_data.record.body.as_ref().unwrap() {
+                                    AnyValue::Int(value) => value.to_string(),
+                                    AnyValue::String(value) => value.to_string(),
+                                    AnyValue::Boolean(value) => value.to_string(),
+                                    AnyValue::Double(value) => value.to_string(),
+                                    AnyValue::Bytes(value) => String::from_utf8_lossy(&value).to_string(),
+                                    AnyValue::ListAny(value) => "".to_string(),
+                                    AnyValue::Map(value) => "".to_string()
+                                },
+                        FieldFormat::Default,
+                    0);
+                        }
+                        if event_id > 0 {
+                            eb.add_value("eventId", event_id, FieldFormat::SignedInt, 0);
+                        }
+                        if event_name.len() > 0 {
+                            eb.add_str("name", event_name, FieldFormat::Default, 0);
+                        }
+                    };
+                }
+
+                // populate CS PartC
+                if log_data.record.attributes.is_some() {
+                    cs_c_count = log_data.record.attributes.as_ref().unwrap().len() as u8 - event_count as u8;
+                }
+                if cs_c_count > 0 {
+                    eb.add_struct("PartC", cs_c_count, 0);
+                    {
+                        self.add_attributes_to_event(&mut eb, &mut log_data.record.attributes.as_ref().unwrap().iter());
+                    }
+                }
+                eb.write(&log_es, None, None);
+            });
+            return Ok(());
+        }
+        Ok(())
+    }
+    
+}
+
+impl Debug for UserEventsExporter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("user_events log exporter")
+    }
+}
+
+#[async_trait]
+impl opentelemetry_sdk::export::logs::LogExporter for UserEventsExporter {
+    async fn export(&mut self, batch: Vec<opentelemetry_sdk::export::logs::LogData>) -> opentelemetry_api::logs::LogResult<()> {
+        for log_data in batch {
+            let _ = self.export_log_data(&log_data);    
+        }
+        Ok(())
+    } 
```

Comment:
Does `shutdown()` also need to be implemented?

Reply:
There is nothing to cleanup in shutdown. The deregister for user_events is automatically handled when EventHeader instance goes out of scope.

Reply:
Interesting. I thought the compiler would complain if we didn't implement all the functions in an inherited trait explicitly.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/export/logs/mod.rs#L16 There is default implementation (empty block), so its optional to implement it.